### PR TITLE
Unit 4 - Meshes

### PR DESCRIPTION
**Pull request by dylanebert**

### CODE REVIEW

File: units/en/unit4/hands-on.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Hands-on
+
+Initially, we planned to walk through the Marching Cubes algorithm and apply it to the [LGM Demo](https://huggingface.co/spaces/dylanebert/LGM-mini). However, recent advancements in mesh generation have made this approach less relevant.
+
+While a deep dive into the methods behind [MeshAnything](https://huggingface.co/spaces/Yiwen-ntu/MeshAnything) would be much more pertitent, its newness and [non-commercial license](https://github.com/buaacyw/MeshAnything/blob/main/LICENSE.txt) make it suboptimal for the time being.
```
Comment (osanseviero) [inline units/en/unit4/hands-on.mdx]:
```suggestion
While a deep dive into the methods behind [MeshAnything](https://huggingface.co/spaces/Yiwen-ntu/MeshAnything) would be much more pertinent, its newness and [non-commercial license](https://github.com/buaacyw/MeshAnything/blob/main/LICENSE.txt) make it suboptimal for the time being.
```

File: units/en/unit4/hands-on.mdx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+Instead, here are some resources based on your goals:
+
+- [Splat to Mesh](https://huggingface.co/spaces/dylanebert/splat-to-mesh): If you followed along with with the LGM-based activities and want to produce the final mesh, this open source demo is based on the original [LGM](https://github.com/3DTopia/LGM) codebase. Note that this method is slow and resource-intensive.
```
Comment (osanseviero) [inline units/en/unit4/hands-on.mdx:9]: Maybe let's switch this to ZeroGPU so it's always on and doesn't go to sleep

File: units/en/unit4/hands-on.mdx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+- [InstantMesh](https://huggingface.co/spaces/TencentARC/InstantMesh): This is fast and state-of-the-art approach uses FlexiCubes to produce the final mesh. It currently ranks at the top of the [3D Arena](https://huggingface.co/spaces/dylanebert/3d-arena) leaderboard.
```
Comment (osanseviero) [inline units/en/unit4/hands-on.mdx]: Not anymore at the top

File: units/en/unit4/hands-on.mdx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+- [meshgpt-pytorch](https://github.com/lucidrains/meshgpt-pytorch): This open source reimplementation of [MeshGPT](https://huggingface.co/papers/2311.15475) provides a good starting point for open-source differentiable mesh generation. [MeshAnything](https://huggingface.co/papers/2311.15475) builds upon MeshGPT.
```
Comment (osanseviero) [inline units/en/unit4/hands-on.mdx]: Maybe mention this does not provide weights

File: units/en/unit4/marching-cubes.mdx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+1. **Polygon Count:** High polygon meshes require more computation for rendering, which can significantly impact performance in real-time applications.
+2. **Edge Flow:** Poor edge flow affects how the mesh deforms when animated, resulting in undesirable artifacts like creasing and pinching.
+3. **Texturing:** The dense and irregular topology complicates UV mapping and texturing, resulting in texture artifacts.
+
+![Mesh Topology](https://huggingface.co/datasets/dylanebert/ml-for-3d-course/resolve/main/topology.jpg)
+
+### Improvements
+
+Techniques like [FlexiCubes](https://research.nvidia.com/labs/toronto-ai/flexicubes/) address some limitations by allowing the mesh vertices to move, creating smoother surfaces. This approach is used by [InstantMesh](https://huggingface.co/spaces/TencentARC/InstantMesh), the current [leading](https://huggingface.co/spaces/dylanebert/3d-arena) open source 3D pipeline. However, the resulting meshes remain overly dense and impractical for production.
```
Comment (osanseviero) [inline units/en/unit4/marching-cubes.mdx]:
```suggestion
Techniques like [FlexiCubes](https://research.nvidia.com/labs/toronto-ai/flexicubes/) address some limitations by allowing the mesh vertices to move, creating smoother surfaces. This approach is used by [InstantMesh](https://huggingface.co/spaces/TencentARC/InstantMesh), the current [leading](https://huggingface.co/spaces/dylanebert/3d-arena) open-source 3D pipeline. However, the resulting meshes remain overly dense and impractical for production.
```

File: units/en/unit4/hands-on.mdx (hunk 3 above)
Comment (osanseviero) [reply units/en/unit4/hands-on.mdx]: Maybe mention open models

**Review by osanseviero — COMMENTED** [review]
Looking great :fire:

File: units/en/unit4/hands-on.mdx (hunk 2 above)
Comment (dylanebert) [reply units/en/unit4/hands-on.mdx:9]: Unfortunately I tried for several hours and failed to get splat-to-mesh working on ZeroGPU :/

Comment (osanseviero) [reply units/en/unit4/hands-on.mdx:9]: I see. Please share which was the issue in the zero GPU channel so we can look into it

**Pull request merged**